design-proposals/monitoring-code-refactor/monitoring-code-refactor.md

```diff
@@ -0,0 +1,214 @@
+# Monitoring code refactor
+
+**Authors:** João Vilaça <jvilaca@redhat.com>
+
+## What
+
+This design document is proposing a code refactor for the monitoring logic in
+all KubeVirt components. The goal is to have a centralized place for the
+creation of monitoring resources, a consistent monitoring package between all
+projects, a code struct that is easy to maintain and evolve, while moving closer
+to the Kubernetes metric implementation style.
+
+## Why
+
+It’s important to refactor the monitoring code to separate monitoring logic from
+business logic. It will make the codebase more modular, readable, and
+maintainable. It will reduce complexity, reduce the risk of introducing errors,
```

Comment:
I don't understand how this proposal is offering to reduce errors, as you reason here.
If I create an alert, that is basically garbage, how this new object validate it?

Reply:
`the risk of introducing errors` here refers to the fact that now that the KubeVirt is in a more advanced stage and we are seeing a considerable increase in monitoring features added across all repositories, having business logic intertwined with monitoring logic, makes it harder to contribute to either one and changes more prone to introduce errors

Reply:
Can you give me an example of such a case where monitoring logic interfered with business logic?

Reply:
https://github.com/kubevirt/kubevirt/pull/7465/files#diff-1baf192bb9c4560d7d40f008b7ba5c680ebce669b8f1b7639445abd3e7269688

that we ended up removing later

Reply:
OK thanks